jbmc/src/java_bytecode/java_entry_point.cpp

```diff
@@ -307,18 +307,25 @@ bool is_java_main(const symbolt &function)
   return named_main && is_static && has_correct_type && public_access;
 }
 
-///  Extends \p init_code with code that allocates the objects used as test
-///  arguments for the function under test (\p function) and
-///  non-deterministically initializes them.
-///
-///  All the code generated by this function goes to __CPROVER__start, just
-///  before the call to the method under test.
-///
-///  \returns A std::vector of symbol_exprt, one per parameter of \p function,
-///  containing the objects that can be used as arguments for \p function.
-exprt::operandst java_build_arguments(
+/// \param function: The function under test, for which to build the arguments.
+/// \param symbol_table: For the purposes of adding any local variables
+///   generated or any functions which are generated and called.
+/// \param assume_init_pointers_not_null: If this is true then any reference
+///   type parameters to the function under tests are assumed to be non-null.
+/// \param object_factory_parameters: Configuration of the object factory.
+/// \param pointer_type_selector: Means of selecting the type of value
+///   constructed for reference types which are initialised by the code
+///   returned.
+/// \returns A pairing of the code to initialise the arguments and a std::vector
+///   of the expressions for these arguments. The vector contains one element
```

Comment:
:bulb: might be good to explicitly state the first parameter will be the `this` parameter if it is a member function.

Reply:
I think the `this` parameter, should be determined by whether `parametert.get_this()` returns `true`. So although the first parameter for a member function would usually be the `this` parameter, it is probably better not to write new code based on that assumption.